GameServer/Logic/GameObjects/AzirTurret.cs

```diff
@@ -0,0 +1,209 @@
+﻿using InibinSharp;
+using LeagueSandbox.GameServer.Core.Logic;
+using LeagueSandbox.GameServer.Core.Logic.RAF;
+using LeagueSandbox.GameServer.Logic.Enet;
+using LeagueSandbox.GameServer.Logic.Maps;
+using LeagueSandbox.GameServer.Logic.Packets;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LeagueSandbox.GameServer.Logic.GameObjects
+{
+    public class AzirTurret : Unit
+    {
+        private const float TURRET_RANGE = 905.0f;
+        private string _name;
+        private Unit _owner;
+        private float _globalGold = 250.0f;
+        private float _globalExp = 0.0f;
+
+        public AzirTurret(Game game, Unit owner, uint id, string name, string model, float x = 0, float y = 0, TeamId team = TeamId.TEAM_BLUE) : base(game, id, model, new Stats(), 50, x, y, 1200)
+        {
+            this._name = name;
+            this._owner = owner;
+
+            buildTurret();
+
+            setTeam(team);
+        }
+
+        public int getEnemyChampionsCount()
+        {
```

Comment:
This function is performance inefficient, as you're looping through the whole champion list through 4 times during this, while it could be done in one loop.

The champions by team probably SHOULD be cached and have a better system overall around it, but that isn't the case yet, so I'd suggest you to add a `GetChampionsByTeam` function to the map object, something along the lines of:
```CS
public Dictionary<string, List<Champion>> GetChampionsByTeam()
{
    var result = new Dictionary<string, List<Champion>>();
    foreach (var kvp in _champions)
    {
        var champion = (Champion)kvp.Value;
        var team = champion.getTeam()
        if(!result.ContainsKey(team))
        {
            result[team] = new List<Champion>();
        }
        result[team].Add(champion);
    }
    return result;
}
```
This way you can simply call that function once and you already have everything you need, you should get the idea. Feel free to ping throw me a message if you need help